src/test/java/io/jenkins/plugins/analysis/warnings/recorder/RecorderITest.java

```diff
@@ -0,0 +1,247 @@
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlFormUtil;
+import com.gargoylesoftware.htmlunit.html.HtmlNumberInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.HealthReport;
+import hudson.model.Project;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Java;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+/**
+ * Verify the Java parser's info page and the health report configuration by the web UI.
+ *
+ * @author Tobias Redl
+ */
+public class RecorderITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private static final int HEALTHY_THRESHOLD = 1;
+    private static final int UNHEALTHY_THRESHOLD = 9;
+    private static final String ENABLED_MESSAGE = "Enabling health report (Healthy=1, Unhealthy=9, Minimum Severity=LOW)";
+    private static final String DISABLED_MESSAGE = "Health report is disabled - skipping";
+
+    private void shouldCreateFreeStyleJobAndVerifyJavaInfoPage(final String warningsFile, final int warningsCount,
+            final int healthScore, final boolean hasHealthReport) throws IOException {
+        final String localFile = "javacWarnings.txt";
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, warningsFile, localFile);
+
+        Java java = new Java();
+        java.setPattern(localFile);
+        enableWarnings(project, java);
+
+        if (hasHealthReport) {
+            new WarningsRecorderConfigurationPageObject(project)
+                    .setHealthyThreshold(HEALTHY_THRESHOLD)
+                    .setUnhealthyThreshold(UNHEALTHY_THRESHOLD)
+                    .submit();
+        }
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        JavaInfoPageObject infoPage = new JavaInfoPageObject(project, 1);
+
+        assertThat(result.getTotalSize()).isEqualTo(warningsCount);
+        assertThat(infoPage.getInformationMessages()).isEqualTo(result.getInfoMessages());
+        assertThat(infoPage.getErrorMessages()).isEqualTo(result.getErrorMessages());
+
+        if (hasHealthReport) {
+            assertThat(infoPage.getInformationMessages()).contains(ENABLED_MESSAGE);
+            HealthReport healthReport = project.getBuildHealth();
+            assertThat(healthReport.getScore()).isEqualTo(healthScore);
+        }
+        else {
+            assertThat(infoPage.getInformationMessages()).contains(DISABLED_MESSAGE);
+        }
+    }
+
+    /**
+     * Verify the Java parser's info page with 0 warnings. Verify the health report (healthy=1, unhealthy=9) configured
+     * by the web UI has a score of 100.
+     *
+     * @throws IOException
```

Comment:
Checked exceptions are evil ;-) Am besten in der Methode fangen und in eine RuntimeException wandeln. Dann ist der JavaDoc einfacher.